doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst

```diff
@@ -0,0 +1,290 @@
+
+.. code-block:: xml
+
+  <depend>moveit_visual_tools</depend>
+
+Then in your ``CMakeLists.xml`` add this line to the section of ``find_package`` statements:
+
+.. code-block:: cmake
+
+  find_package(moveit_visual_tools REQUIRED)
+
+Further down in the file extend the ``ament_target_dependencies`` macro call to include the new dependency like this:
+
+.. code-block:: cmake
+
+  ament_target_dependencies(
+    hello_moveit
+    "moveit_ros_planning_interface"
+    "moveit_visual_tools"
+    "rclcpp"
+  )
+
+To verify you added the dependency correctly add the required include to your source file ``hello_moveit.cpp``:
+
+.. code-block:: C++
+
+  #include <moveit_visual_tools/moveit_visual_tools.h>
+
+To test that this all worked open a terminal in the workspace directory (remembering to source your ros install in opt) and then build with colcon:
+
+.. code-block:: bash
+
+  cd ~/ws_moveit2
+  colcon build --mixin debug
+
+2 Create a ROS executor and spin the node on a thread
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Before we can initialize MoveItVisualTools we need to have a executor spinning on our ROS node.
+This is necessary because of how MoveItVisualTools interacts with ROS services and topics.
+
+.. code-block:: C++
+
+  #include <thread>  // <---- add this to the set of includes at the top
+
+    ...
+
+    // Create a ROS logger
+    auto const logger = rclcpp::get_logger("hello_moveit");
+
+    // We spin up a SingleThreadedExecutor so MoveItVisualTools interact with ROS
+    rclcpp::executors::SingleThreadedExecutor executor;
+    executor.add_node(node);
+    auto spinner = std::thread([&executor]() { executor.spin(); });
+
+    // Create the MoveIt MoveGroup Interface
+    ...
+
+    // Shutdown ROS
+    rclcpp::shutdown();  // <--- This will cause the spin function in the thread to return
+    spinner.join();  // <--- Join the thread before exiting
+    return 0;
+  }
+
+After each one of these changes you should rebuild your workspace to make sure you don't have any syntax errors.
+
+3 Create and Initialize MoveItVisualTools
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Next we will construct and initialize MoveItVisualTools after the construction of MoveGroupInterface.
+
+.. code-block:: C++
+
+    // Create the MoveIt MoveGroup Interface
+    using moveit::planning_interface::MoveGroupInterface;
+    auto move_group_interface = MoveGroupInterface(node, "panda_arm");
+
+    // Construct and initialize MoveItVisualTools
+    auto moveit_visual_tools = moveit_visual_tools::MoveItVisualTools{
+        node, "panda_link0", rviz_visual_tools::RVIZ_MARKER_TOPIC,
+        move_group_interface.getRobotModel()};
+    moveit_visual_tools.deleteAllMarkers();
+    moveit_visual_tools.loadRemoteControl();
+
+In the constructor we pass in the ROS node, the base link of the robot, the marker topic to use (more on this later), and the robot model (which we get from the move_group_interface).
```

Comment:
This sentence reads a bit awkwardly. I think it's missing a comma, but even so, it sounds a bit Yoda-like. Not sure that my suggestion below is the best wording, but I'd re-word it one way or another.

```suggestion
We pass the following into the constructor: the ROS node, the base link of the robot, the marker topic to use (more on this later), and the robot model (which we get from the move_group_interface).
```